### PR TITLE
add arg parser for server and client

### PR DESCRIPTION
**Pull request by lyppg**

### CODE REVIEW

File: src/config.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#ifndef LIBCONFIG_H
+#define LIBCONFIG_H
+
+typedef struct ServerConfig {
+    int data_port;
+    int control_port;
```
Comment (thesues) [inline src/config.h:6]: manage_port

File: src/config.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    std::string log_level;
+    size_t prealloc_size;
+} server_config_t;
+
+typedef struct ClientConfig {
+    int data_port;
+    int control_port;
+    std::string log_level;
+    std::string connect_host;
```
Comment (thesues) [inline src/config.h:15]: ipaddr

File: src/config.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#ifndef LIBCONFIG_H
+#define LIBCONFIG_H
+
+typedef struct ServerConfig {
+    int data_port;
+    int control_port;
+    std::string log_level;
+    size_t prealloc_size;
```
Comment (thesues) [inline src/config.h:8]: add more comment, prealloc_size's unit is GB

File: src/config.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+} server_config_t;
+
+typedef struct ClientConfig {
+    int data_port;
+    int control_port;
```
Comment (thesues) [inline src/config.h:13]: manage_port

File: infinity/test_infinity.py [hunk 5]
```diff
@@ -8,6 +8,7 @@
 import random
 import string
 import contextlib
+from infinity._infinity import ClientConfig
```
Comment (thesues) [inline infinity/test_infinity.py:11]: from python, do not import ClientConfig from .so, check __init__.py

File: infinity/test_infinity.py [hunk 6]
```diff
@@ -45,11 +46,15 @@ def get_gpu_count():
 @pytest.mark.parametrize("new_connection", [True, False])
 @pytest.mark.parametrize("local", [True, False])
 def test_basic_read_write_cache(server, dtype, new_connection, local):
-    conn = infinity.InfinityConnection()
+    config = ClientConfig()
+    config.service_port = 22345
+    config.host_addr = "10.192.24.218"
```
Comment (thesues) [inline infinity/test_infinity.py:51]: ??this is unit test.  only '127.0.0.1' is allowed

Comment (thesues) [reply infinity/test_infinity.py:51]: could use a global Client/Server config.

File: infinity/lib.py [hunk 7]
```diff
@@ -92,25 +92,21 @@ class InfinityConnection:
     OP_RDMA_WRITE = "D"
     OP_RDMA_READ = "A"
 
-    def __init__(
-        self,
-    ):
+    def __init__(self, config):
```
Comment (thesues) [inline infinity/lib.py:95]: if we have a config, config will tell this is RDMA or Local. so we do not need local_connect any more

File: infinity/test.py [hunk 8]
```diff
@@ -31,10 +32,14 @@ def run(conn):
...
+    config.service_port = 22345
+    config.host_addr = "10.192.24.216"
+
+    rdma_conn = InfinityConnection(config)
+    rdma_conn.connect()
     run(rdma_conn)
 
-    local_conn = InfinityConnection()
+    local_conn = InfinityConnection(config)
     local_conn.local_connect()
```
Comment (thesues) [inline infinity/test.py:44]: maybe we could delete `local_connect`

File: infinity/lib.py (hunk 7 above)
Comment (lyppg) [reply infinity/lib.py:97]: > if we have a config, config will tell this is RDMA or Local. so we do not need local_connect any more

如果127.0.0.1是local，那单元测试，127.0.0.1就自动走local了，测不了rdma了。我提交修改了

File: infinity/test_infinity.py (hunk 6 above)
Comment (lyppg) [reply infinity/test_infinity.py]: fixed

File: infinity/test_infinity.py (hunk 5 above)
Comment (lyppg) [reply infinity/test_infinity.py]: fixed

**Pull request closed without merging**